File: exercism/python/space-age/space_age.py
class SpaceAge:
    earth_years = {
        'mercury': 0.2408467,
        'venus': 0.61519726,
        'earth': 1.0,
        'mars': 1.8808158,
        'jupiter': 11.862615,
        'saturn': 29.447498,
        'uranus': 84.016846,
        'neptune': 164.79132,
    }

    earth_seconds = 31557600


    def __init__(self, seconds):
        self.age_seconds = seconds
        self.earth_year = round(self.age_seconds / SpaceAge.earth_seconds, 2)

        for planet, p_years in SpaceAge.earth_years.items():
            earth_years = lambda p_years=p_years: round(self.earth_year / p_years, 2)
            setattr(self, f'on_{planet}', earth_years)

File: exercism/go/space-age/space_age.go
package space

type Planet string

const SECONDS_TO_EARTH_YEAR = 31557600

var orbital_periods = map[Planet]float64{
	"Mercury": 0.2408467,
	"Venus":   0.61519726,
	"Earth":   1.0,
	"Mars":    1.8808158,
	"Jupiter": 11.862615,
	"Saturn":  29.447498,
	"Uranus":  84.016846,
	"Neptune": 164.79132,
}

func Age(age float64, planet Planet) float64 {
	earth_years := age / SECONDS_TO_EARTH_YEAR
	return earth_years / orbital_periods[planet]
}
